### PR TITLE
#88962 Ensure Cache-Clearing Works w/ Persistent Object Cache

### PR DESCRIPTION
**Pull request by ggwicz**
**Ticket:** [**#88962**](http://central.tri.be/issues/88962)

### CODE REVIEW

File: includes/pardot-plugin-class.php [hunk 1]
```diff
@@ -1094,11 +1112,124 @@ static function get_cache( $key ) {
...
-		wp_cache_set( $key, $value, 'pardot' );
-		if ( $set_transient )
-			set_transient( "pardot_{$key}", $value, self::$cache_timeout );
+
+		if ( wp_cache_set( $key, $value, 'pardot' ) ) {
+			self::save_cache_key( $key );
+		}
+		
+		if ( $set_transient ) {
+			if ( set_transient( "pardot_{$key}", $value, self::$cache_timeout ) ) {
```
Comment (bordoni) [inline includes/pardot-plugin-class.php]: Can we merge these two conditionals in one? Or invert the first one with a `return`? So we can avoid indentation.

File: includes/pardot-plugin-class.php [hunk 2]
```diff
@@ -38,6 +38,20 @@ class Pardot_Plugin {
 	 */
 	static $cache_timeout = PARDOT_API_CACHE_TIMEOUT;
 
+	/**
+	 * @var The name of the wp_option where cache keys will be stored.
+	 *
+	 * @since TBD
+	 */
+	static $saved_cache_keys     = '_pardot_cache_keys';
```
Comment (bordoni) [inline includes/pardot-plugin-class.php]: We should declare the visibility of this.

File: includes/pardot-plugin-class.php [hunk 3]
```diff
@@ -38,6 +38,20 @@ class Pardot_Plugin {
...
+
+	/**
+	 * @var The name of the wp_option where transient keys will be stored.
+	 *
+	 * @since TBD
+	 */
+	static $saved_transient_keys = '_pardot_transient_keys';
```
Comment (bordoni) [inline includes/pardot-plugin-class.php]: We should declare the visibility of this.

File: includes/pardot-plugin-class.php [hunk 4]
```diff
@@ -1094,11 +1112,124 @@ static function get_cache( $key ) {
...
+	}
+
+	/**
+	 * Stores the name of a transient key so that we can reference it later in bulk-deletion.
+	 *
+	 * @since TBD
+	 *
+	 * @return array
+	 */
+	static function save_transient_key( $key ) {
```
Comment (bordoni) [inline includes/pardot-plugin-class.php]: We should declare the visibility of this.

File: includes/pardot-plugin-class.php [hunk 5]
```diff
@@ -1094,11 +1112,124 @@ static function get_cache( $key ) {
...
+	}
+
+	/**
+	 * Stores the name of a wp_cache key so that we can reference it later in bulk-deletion.
+	 *
+	 * @since TBD
+	 *
+	 * @return array
+	 */
+	static function save_cache_key( $key ) {
```
Comment (bordoni) [inline includes/pardot-plugin-class.php]: We should declare the visibility of this.

File: includes/pardot-plugin-class.php [hunk 6]
```diff
@@ -1094,11 +1112,124 @@ static function get_cache( $key ) {
...
 	}
 
+	/**
+	 * Get saved transient keys.
+	 *
+	 * @since TBD
+	 *
+	 * @return array
+	 */
+	static function get_saved_transient_keys() {
```
Comment (bordoni) [inline includes/pardot-plugin-class.php]: We should declare the visibility of this.

File: includes/pardot-plugin-class.php [hunk 7]
```diff
@@ -1094,11 +1112,124 @@ static function get_cache( $key ) {
...
+	}
+
+	/**
+	 * Get saved cache keys.
+	 *
+	 * @since TBD
+	 *
+	 * @return array
+	 */
+	static function get_saved_cache_keys() {
```
Comment (bordoni) [inline includes/pardot-plugin-class.php]: We should declare the visibility of this.

File: includes/pardot-plugin-class.php [hunk 8]
```diff
@@ -1094,11 +1112,124 @@ static function get_cache( $key ) {
...
+	}
+
+	/**
+	 * Totally clears out all Pardot transients and wp_cache items.
+	 *
+	 * @since TBD
+	 *
+	 * @return bool True if the cache was cleared, false otherwise.
+	 */	
+	static function clear_cache() {
```
Comment (bordoni) [inline includes/pardot-plugin-class.php]: We should declare the visibility of this.

**Comment by ggwicz** [comment]
I've implementing @bordoni's feedback; this one's ready for another review! 🔍

**Pull request merged**